bears/general/CasingBear.py

```diff
@@ -0,0 +1,153 @@
+import re
+
+from coalib.bears.LocalBear import LocalBear
+from coalib.bearlib.naming_conventions import (
+    to_camelcase, to_pascalcase, to_snakecase)
+from bears.general.AnnotationBear import AnnotationBear
+from coalib.bearlib.languages.LanguageDefinition import LanguageDefinition
+from coalib.results.SourceRange import SourceRange
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+
+
+class CasingBear(LocalBear):
+
+    def run(self,
+            filename,
+            file,
+            dependency_results,
+            casing: str,
+            language: str):
+        """
+        Checks whether all identifier names (variables, classes, objects)
+        follow a certain naming convention.
+
+        :param filename:           Name of the file that needs to be checked.
+        :param file:               File that needs to be checked in the form
+                                   of a list of strings.
+        :param dependency_results: A dict of dependencies with bear name as
+                                   key and results as value
+        :param casing:             camelCasing or snake_casing or PascalCasing
+        :param language:           The language of the file, which is used to
+                                   determine the keywords to ignore.
+        """
+        casing_convention = {"camel": to_camelcase,
+                             "pascal": to_pascalcase,
+                             "snake": to_snakecase}
+
+        if casing not in casing_convention:
+            self.err("Invalid casing convention provided: " + casing)
+            return
+
+        self.convertor = casing_convention[casing]
+        coalang = LanguageDefinition(language)
+
+        # TODO: Adapt after coala-analyzer/coala/issues/2200 is solved
+        if "keywords" not in coalang or "special_chars" not in coalang:
+            self.warn("CasingBear did not run because 'keywords' and "
+                      "'special_chars' are necessary fields in the "
+                      "coalang definition for " + language)
```

Comment:
i still think this can better IMO, you tell the user that keywords and special_chars are necessary fields, but you didnt tell them that they weren't found